packages/ui/src/components/pressable/pressable.build-press-effect-styles.native.ts

```diff
@@ -0,0 +1,50 @@
+import { ViewStyle } from 'react-native';
+import {
+  CSSTransitionDelay,
+  CSSTransitionDuration,
+  CSSTransitionTimingFunction,
+} from 'react-native-reanimated';
+
+import { isEmpty } from 'remeda';
+
+import { ResolvedPressEffect, pressEffectProperties } from './pressable.types.native';
+
+const defaultDuration = 150;
+
+interface PressEffectStyle extends ViewStyle {
+  transitionProperty?: string[];
+  transitionDuration?: CSSTransitionDuration[];
+  transitionTimingFunction?: CSSTransitionTimingFunction[];
+  transitionDelay?: CSSTransitionDelay[];
+}
+
+export function buildPressEffectStyles(
+  pressEffect: ResolvedPressEffect | null,
+  isPressed: boolean
+): PressEffectStyle {
+  if (!pressEffect || isEmpty(pressEffect)) return {};
+
+  const style: PressEffectStyle = {};
+  const transitionProperty: string[] = [];
+  const transitionDuration: CSSTransitionDuration[] = [];
+  const transitionTimingFunction: CSSTransitionTimingFunction[] = [];
+  const transitionDelay: CSSTransitionDelay[] = [];
+
+  for (const property of pressEffectProperties) {
+    const config = pressEffect[property];
+    if (!config) continue;
+
+    (style[property] as typeof config.to) = isPressed ? config.to : config.from;
+    transitionProperty.push(property);
+    transitionDuration.push(config.duration ?? defaultDuration);
+    transitionTimingFunction.push(config.timing ?? 'ease');
+    transitionDelay.push(config.delay ?? 0);
+  }
```

Comment:
I'd suggest using remeda instead of for...loops:

```suggestion
  R.pipe(
    R.entries(pressEffect),
    R.map(([property, config]) => {
      (style[property] as typeof config.to) = isPressed ? config.to : config.from;
    transitionProperty.push(property);
    transitionDuration.push(config.duration ?? defaultDuration);
    transitionTimingFunction.push(config.timing ?? 'ease');
    transitionDelay.push(config.delay ?? 0);
    }),
  );
```

It also should be possible to skip mutating objects if we return values in the map here and then use fromEntries to reconstruct the object from the array, but that might be too much remeda :D

Reply:
I actually had a little more elaborate fluent expression here, as the initial draft supported many more properties I ended up removing. Remeda is great for declarative transformations, but this is one of the places where it improve much if at all. I wouldn't worry about small local mutations either--they're harmless.